app/views/export/_ad_hoc_element.html.erb

```diff
@@ -0,0 +1,23 @@
+<% if ad_hoc_element.is_a? Array %>
+  <div>
+    <%= render partial: "export/ad_hoc_element", collection: ad_hoc_element %>
+  </div>
+<% elsif ad_hoc_element.is_a? Hash %>
+  <% if ad_hoc_element["type"] == "email" %>
+    <h2>Email: <%= ad_hoc_element["subject"] %></h2>
+    <%= ad_hoc_element["value"].try(:html_safe) %>
+  <% elsif ad_hoc_element["type"] == "adhoc-label" %>
+    <b><%= ad_hoc_element["value"].try(:html_safe) %></b>
+  <% elsif ad_hoc_element["type"] == "text" %>
+    <%= ad_hoc_element["value"].try(:html_safe) %>
+  <% elsif ad_hoc_element["type"] == "attachments" %>
+    <%# attachments are exported elsewhere %>
+  <% elsif ad_hoc_element["type"] == "checkbox" %>
+    <% checked_str = ad_hoc_element["answer"] ? "checked" : "" %>
+    <p><input disabled type="checkbox" <%= checked_str %>> <%= ad_hoc_element["value"].try(:html_safe) %></p>
+  <% else %>
+    <% raise Exception.new("strange ad-hoc content: #{ad_hoc_element}") %>
+  <% end %>
+<% else %>
+  <% raise Exception.new("strange ad-hoc card") %>
```

Comment:
is there other context that would identify the oddball card?

Reply:
It's really just to throw an error if we are looking at content that I didn't account for. I've exported a fair number of papers and they all seem fine, so it seems unlikely we'll hit these errors.